drv/lpc55-swd/src/main.rs

```diff
@@ -1058,21 +1297,626 @@ impl ServerImpl {
         }
 
         // Reset the STM32, causing it to reboot into the newly-set slot
-        use drv_lpc55_gpio_api::{Pins, Value};
-        let gpio = Pins::from(self.gpio);
-        gpio.set_val(ROT_TO_SP_RESET_L, Value::Zero);
+        self.sp_reset(false);
+        Ok(())
+    }
+
+    fn sp_reset(&mut self, squelch_notify: bool) {
+        self.sp_reset_enter();
         hl::sleep_for(10);
-        gpio.set_val(ROT_TO_SP_RESET_L, Value::One);
+        self.sp_reset_leave(squelch_notify);
+    }
+
+    fn sp_reset_enter(&mut self) {
+        setup_rot_to_sp_reset_l_out(self.gpio);
+        let gpio = Pins::from(self.gpio);
+        gpio.set_val(ROT_TO_SP_RESET_L_OUT, Value::Zero);
+    }
+
+    /// Cleanup is true if we don't want to be notified of this reset
+    fn sp_reset_leave(&mut self, squelch_notify: bool) {
+        let gpio = Pins::from(self.gpio);
+        setup_rot_to_sp_reset_l_in(self.gpio);
+        gpio.set_val(ROT_TO_SP_RESET_L_IN, Value::One); // should be a no-op
+        if squelch_notify {
```

Comment:
I'm confused by `squelch_notify` here; wouldn't the IRQ have already fired in `sp_reset_enter`?

Reply:
Doesn't the kernel queue the notification as soon as the pin changes state?  I would expect that we need a call to `sys_irq_control_clear_pending` here as well.

Reply:
You are correct. Moved the extra interrupt suppression into the notification handler which is the only place that generates them.